xsnippet_api/resources/snippets.py

```diff
@@ -10,11 +10,206 @@
 
 
+# List of supported syntaxes. So far it's used only for validation purpose and
+# taking into account that syntax highlighting is going to be implemented on
+# client's side, we need to start consider removing that enum.
+_SYNTAXES = [
+    '1c',
+    'abnf',
+    'accesslog',
+    'actionscript',
+    'ada',
+    'apache',
+    'applescript',
+    'arduino',
+    'armasm',
+    'asciidoc',
+    'aspectj',
+    'autohotkey',
+    'autoit',
+    'avrasm',
+    'awk',
+    'axapta',
+    'bash',
+    'basic',
+    'bnf',
+    'brainfuck',
+    'cal',
+    'c',
+    'capnproto',
+    'ceylon',
+    'clojure',
+    'clojure-repl',
+    'cmake',
+    'coffeescript',
+    'coq',
+    'cos',
+    'cpp',
+    'crmsh',
+    'crystal',
+    'cs',
+    'csp',
+    'css',
+    'd',
+    'dart',
+    'delphi',
+    'diff',
+    'django',
+    'dns',
+    'dockerfile',
+    'dos',
+    'dsconfig',
+    'dts',
+    'dust',
+    'ebnf',
+    'elixir',
+    'elm',
+    'erb',
+    'erlang-repl',
+    'erlang',
+    'excel',
+    'fix',
+    'fortran',
+    'fsharp',
+    'gams',
+    'gauss',
+    'gcode',
+    'gherkin',
+    'glsl',
+    'go',
+    'golo',
+    'gradle',
+    'groovy',
+    'haml',
+    'handlebars',
+    'haskell',
+    'haxe',
+    'hsp',
+    'html',
+    'htmlbars',
+    'http',
+    'inform7',
+    'ini',
+    'irpf90',
+    'java',
+    'javascript',
+    'json',
+    'julia',
+    'kotlin',
+    'lasso',
+    'ldif',
+    'less',
+    'lisp',
+    'livecodeserver',
+    'livescript',
+    'lsl',
+    'lua',
+    'makefile',
+    'markdown',
+    'mathematica',
+    'matlab',
+    'maxima',
+    'mel',
+    'mercury',
+    'mipsasm',
+    'mizar',
+    'mojolicious',
+    'monkey',
+    'moonscript',
+    'nginx',
+    'nimrod',
+    'nix',
+    'nsis',
+    'objectivec',
+    'ocaml',
+    'openscad',
+    'oxygene',
+    'parser3',
+    'perl',
+    'pf',
+    'php',
+    'pony',
+    'powershell',
+    'processing',
+    'profile',
+    'prolog',
+    'protobuf',
+    'puppet',
+    'purebasic',
+    'python',
+    'q',
+    'qml',
+    'r',
+    'rib',
+    'roboconf',
+    'rsl',
+    'ruby',
+    'ruleslanguage',
+    'rust',
+    'scala',
+    'scheme',
+    'scilab',
+    'scss',
+    'smali',
+    'smalltalk',
+    'sml',
+    'sqf',
+    'sql',
+    'stan',
+    'stata',
+    'step21',
+    'stylus',
+    'subunit',
+    'swift',
+    'taggerscript',
+    'tap',
+    'tcl',
+    'tex',
+    'thrift',
+    'tp',
+    'twig',
+    'typescript',
+    'vala',
+    'vbnet',
+    'vbscript-html',
+    'vbscript',
+    'verilog',
+    'vhdl',
+    'vim',
+    'x86asm',
+    'xl',
+    'xml',
+    'xquery',
+    'yaml',
+    'zephir',
+]
+
+_SCHEMA = {
+    'title': {'type': 'string'},
+    'content': {'type': 'string', 'required': True},
+    'syntax': {'type': 'string', 'allowed': _SYNTAXES},
+    'tags': {'type': 'list', 'schema': {'type': 'string', 'regex': '[\w_-]+'}},
+    'is_public': {'type': 'boolean', },
```

Comment:
nit: trailing comma in the literal is inconsistent with other values of this dict